Index: coremark-1.01/linux/core_portme.mak
===================================================================
--- coremark-1.01.orig/linux/core_portme.mak
+++ coremark-1.01/linux/core_portme.mak
@@ -8,7 +8,7 @@ OUTFLAG= -o
 CC = gcc
 # Flag: CFLAGS
 #	Use this flag to define compiler options. Note, you can add compiler options from the command line using XCFLAGS="other flags"
-PORT_CFLAGS = -O2
+PORT_CFLAGS = -O3
 FLAGS_STR = "$(PORT_CFLAGS) $(XCFLAGS) $(XLFLAGS) $(LFLAGS_END)"
 CFLAGS = $(PORT_CFLAGS) -I$(PORT_DIR) -I. -DFLAGS_STR=\"$(FLAGS_STR)\"
 #Flag: LFLAGS_END
Index: coremark-1.01/linux64/core_portme.mak
===================================================================
--- coremark-1.01.orig/linux64/core_portme.mak
+++ coremark-1.01/linux64/core_portme.mak
@@ -8,7 +8,7 @@ OUTFLAG= -o
 CC = gcc
 # Flag: CFLAGS
 #	Use this flag to define compiler options. Note, you can add compiler options from the command line using XCFLAGS="other flags"
-PORT_CFLAGS = -O2
+PORT_CFLAGS = -O3
 FLAGS_STR = "$(PORT_CFLAGS) $(XCFLAGS) $(XLFLAGS) $(LFLAGS_END)"
 CFLAGS = $(PORT_CFLAGS) -I$(PORT_DIR) -I. -DFLAGS_STR=\"$(FLAGS_STR)\"
 #Flag: LFLAGS_END
