--- a/programs/pluto/ipsec_doi.c	2019-04-04 13:49:32.026550608 -0700
+++ openswan-2.6.51.3/programs/pluto/ipsec_doi.c	2019-04-04 13:56:17.622538600 -0700
@@ -425,6 +425,7 @@
 lset_t
 update_policy_from_state(const struct state *st, lset_t policy)
 {
+    policy = policy | (st->st_connection->policy & ~POLICY_IKEV2_MASK);
     if (st->st_pfs_group != NULL)
         policy |= POLICY_PFS;
     if (st->st_ah.present)
