diff -Nur a/linux/net/ipsec/ipsec_xmit.c openswan-2.6.51.3/linux/net/ipsec/ipsec_xmit.c
--- a/linux/net/ipsec/ipsec_xmit.c	2021-08-02 20:40:23.137419083 +0530
+++ openswan-2.6.51.3/linux/net/ipsec/ipsec_xmit.c	2021-10-06 13:44:42.733765958 +0530
@@ -2659,6 +2659,8 @@
 			ip_fast_csum((unsigned char *)ip_hdr(ixs->skb), ip_hdr(ixs->skb)->ihl);
 	}

+	memset(ixs->skb->cb, 0, sizeof(struct ipsec_skb_cb));
+
 	KLIPS_PRINT(debug_tunnel & DB_TN_XMIT,
 		    "klips_debug:ipsec_xmit_send: "
 		    "...done, calling ip_send() on device:%s%s\n",

diff -Nur a/linux/net/ipsec/ipsec_rcv.c openswan-2.6.51.3/linux/net/ipsec/ipsec_rcv.c
--- a/openswan-2.6.51.3/linux/net/ipsec/ipsec_rcv.c	2021-10-06 16:35:51.342207986 +0530
+++ openswan-2.6.51.3/linux/net/ipsec/ipsec_rcv.c	2021-10-06 16:33:20.253697995 +0530
@@ -1635,6 +1635,7 @@
 	}

 	/* zero any options that there might be */
+	memset(skb->cb, 0, sizeof(struct ipsec_skb_cb));
 	memset(&(IPCB(skb)->opt), 0, sizeof(struct ip_options));

 	if (debug_rcv)
