--- accel-pptp-0.8.5/pppd_plugin/src/pppd/pppd.h	2010-10-06 18:57:46.000000000 +0530
+++ accel-pptp-0.8.5/pppd_plugin/src/pppd/pppd.h	2018-06-27 10:47:48.258691461 +0530
@@ -54,7 +54,7 @@
 #include <sys/param.h>		/* for MAXPATHLEN and BSD4_4, if defined */
 #include <sys/types.h>		/* for u_int32_t, if defined */
 #include <sys/time.h>		/* for struct timeval */
-#include <net/ppp_defs.h>
+#include <linux/ppp_defs.h>
 
 #if defined(__STDC__)
 #include <stdarg.h>
